pkg/globalnet/controllers/gateway_monitor.go

```diff
@@ -161,10 +161,12 @@ func (g *gatewayMonitor) handleCreatedOrUpdatedEndpoint(obj runtime.Object, numR
 			return false
 		}
 
+		g.syncMutex.Lock()
```

Comment:
This lock is intended to protect the `isGatewayNode` flag so I'm wary of also using it in a disparate manner. This is a case where the standalone [synchronized set](https://github.com/submariner-io/admiral/pull/557/commits/05d6c279d17e1818f0d861803567d36c4a1ae667) is useful.

Reply:
How about https://github.com/submariner-io/submariner/pull/2325 instead, for `isGatewayNode`? `syncMutex` is also used around `g.stopControllers()` in `Stop()`, perhaps that should be moved into `stopControllers()` itself.

Reply:
OK, I’ll drop the synchronization around those.